src/ElasticaService.php

```diff
@@ -391,14 +382,16 @@ public function define($recreate = false)
      */
     public function refresh()
     {
-        $reading_mode = Versioned::get_reading_mode();
-        Versioned::set_reading_mode('Stage.Live');
+        Versioned::withVersionedMode(function () {
+            Versioned::set_stage(Versioned::LIVE);
 
-        foreach ($this->getIndexedClasses() as $class) {
-            //Only index types (or classes) that are not just supporting other index types
-            if (!Config::inst()->get($class, 'supporting_type')) {
-                /** @var DataObject $record */
-                foreach ($class::get() as $record) {
+            foreach ($this->getIndexedClasses() as $class) {
+                //Only index types (or classes) that are not just supporting other index types
+                if (Config::inst()->get($class, 'supporting_type')) {
```

Comment:
I grepped the codebase, is `supporting_type` documented anywhere?

Reply:
Can't see any docs, I'll add some :)

Reply:
Done